CosmosDb.GraphAPI.Recommender/GraphDBHelper.cs

```diff
@@ -0,0 +1,202 @@
+
+        public static async Task<DocumentCollection> CreateCollectionAsync(DocumentClient documentClient, 
+            Database database, string collectionId, string partitionKey, int throughput, bool isPartitionedGraph)
+        {
+            var collection = new DocumentCollection
+            {
+                Id = collectionId
+            };
+
+            if (isPartitionedGraph)
+            {
+                if (string.IsNullOrWhiteSpace(partitionKey))
+                {
+                    throw new ArgumentNullException("PartionKey can't be null for a partitioned collection");
+                }
+
+                collection.PartitionKey.Paths.Add("/" + partitionKey);
+            }
+
+            return await documentClient.CreateDocumentCollectionAsync(
+                database.SelfLink,
+                collection,
+                new RequestOptions { OfferThroughput = throughput });
+        }
+
+        public static async Task DeleteCollectionAsync(DocumentClient documentClient, DocumentCollection documentCollection)
+        {
+            if (documentCollection != null)
+            {
+                await documentClient.DeleteDocumentCollectionAsync(documentCollection.SelfLink);
+            }
+        }
+
+
+        public static async Task<DocumentCollection> GetCollectionAsync(DocumentClient documentClient, 
+            string databaseId, string collectionId)
+        {
+            return await documentClient.ReadDocumentCollectionAsync(
+                UriFactory.CreateDocumentCollectionUri(databaseId, collectionId));
+        }
+
+        // 
+        
+        public static async Task<GraphBulkImport> CreateAndInitGraphImporterAsync(DocumentClient documentClient,
+            string databaseId, DocumentCollection collection)
+        {
+            var graphBulkImport = new GraphBulkImport(documentClient, collection, useFlatProperty: false);
+            await graphBulkImport.InitializeAsync();
+            return graphBulkImport;
+        }
+
+        public static async Task<GraphBulkImportResponse> AddVerticesAsync(GraphBulkImport graphImporter, 
+            IEnumerable<Vertex> vertices)
+        {
+            return await graphImporter.BulkImportVerticesAsync(
+                vertices: vertices,
+                enableUpsert: true);
+        }
+
+        public static async Task<GraphBulkImportResponse> AddEdgesAsync(GraphBulkImport graphImporter, 
+            IEnumerable<Edge> edges)
+        {
+            return await graphImporter.BulkImportEdgesAsync(
+                edges: edges,
+                enableUpsert: true);
+        }
+
+
+        // Genearting
+
+        public static IEnumerable<Vertex> GenerateBrandVertices(List<Brand> brands, string partitionKey)
+        {
+            foreach (var brand in brands)
+            {
+                var vertex = new Vertex(brand.Id.ToString(), "brand");
+                vertex.AddProperty(new VertexProperty(partitionKey, brand.Id.ToString()));
```

Comment:
I don't think we need to use unique values for partitionKey.

This property is intended for scalability and more likely a container.

Take a look at this post and let me know if I'm wrong: https://blog.maartenballiauw.be/post/2012/10/08/what-partitionkey-and-rowkey-are-for-in-windows-azure-table-storage.html